appinventor/appengine/src/com/google/appinventor/client/editor/simple/palette/SimplePaletteItem.java

```diff
@@ -127,7 +127,7 @@ public MockComponent createMockComponent() {
     cacheInternalComponentPrototype();
 
     MockComponent returnedComponentPrototype = componentPrototype;
-    componentPrototype = null;
+    // componentPrototype = null;
```

Comment:
I know this code does not 100% work (it stops creating some new instances), but just sharing it as something's off with the actual component instantiation. 🙂  
Every time a component gets dragged and dropped, just two instances are created but just one fires its `delete`.